File: football-gui-game/src/main/java/com/github/lipinskipawel/controller/PitchController.java
package com.github.lipinskipawel.controller;

import com.github.lipinskipawel.gui.RenderablePoint;

interface PitchController {

    void leftClick(final RenderablePoint renderablePoint);

    void rightClick(final RenderablePoint renderablePoint);

    void tearDown();
}

File: app/src/main/kotlin/com/github/lipinskipawel/controller/PitchController.kt
package com.github.lipinskipawel.controller

import com.github.lipinskipawel.gui.RenderablePoint
import io.github.lipinskipawel.board.engine.Move

interface PitchController {

    fun leftClick(renderablePoint: RenderablePoint)

    fun rightClick(renderablePoint: RenderablePoint)

    fun tearDown()

    fun gameMoves(): List<Move>
}